gitlab/data_source_gitlab_branch.go

```diff
@@ -0,0 +1,74 @@
+package gitlab
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func dataSourceGitlabBranch() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGitlabBranchRead,
```

Comment:
Please use the context-aware CRUD functions, see the [upgrade guide](https://www.terraform.io/plugin/sdkv2/guides/v2-upgrade-guide#more-support-for-context-context).

Also make sure to use the new `ctx` to pass it to the `gitlab` API functions as `gitlab.WithContext(ctx)`. It's usually the last parameter of the API functions.